src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -816,27 +829,57 @@ public void postBlUsersLogin(boolean expandPerms, List<String> include, String u
           .ifPresent(header -> headers.put(HttpHeaders.USER_AGENT, header));
         Optional.ofNullable(xForwardedFor)
           .ifPresent(header -> headers.put(X_FORWARDED_FOR_HEADER, header));
-        loginResponse[0] = client.request(HttpMethod.POST, entity, moduleURL, headers);
```

Comment:
Why not keep this way of making the login request? Then you don't need to do `thenAccept` and `exceptionally` and all of that. Also, why are you creating the second instance of the client? Can't you just reuse the same client as the previous code reused it?

Reply:
Unfortunately, no. The client is created using tenantid, and we have to create a client to call the users API with tenantId, which was in the login API response. The tenantId for the login client and for the user client may be different.